ms-common-auth/src/main/java/net/trajano/ms/authz/ClientCheckResource.java

```diff
@@ -0,0 +1,66 @@
+package net.trajano.ms.authz;
+
+import java.net.URI;
+
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.swagger.annotations.Api;
+import net.trajano.ms.auth.spi.ClientValidator;
+import net.trajano.ms.auth.util.HttpAuthorizationHeaders;
+import net.trajano.ms.authz.internal.TokenCache;
+import net.trajano.ms.authz.internal.TokenCacheEntry;
+import net.trajano.ms.core.ErrorResponses;
+
+/**
+ * Client check resource.
+ *
+ * @author Archimedes Trajano
+ */
+@Api
+@Component
+@Path("/check")
+@PermitAll
+public class ClientCheckResource {
+
+    @Autowired
+    private ClientValidator clientValidator;
+
+    @Autowired
+    private TokenCache tokenCache;
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response validateClient(final ClientCheckRequest req) {
+
+        boolean validClient = false;
+        switch (HttpAuthorizationHeaders.getAuthorizationType(req.getAuthorization())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)